pennylane/devices/default_qubit.py

```diff
@@ -202,21 +202,49 @@ def apply(self, operations, rotations=None, **kwargs):
         rotations = rotations or []
 
         # apply the circuit operations
+        wires_visited = set()
```

Comment:
This is something that struck me as odd when I saw it but currently pennylane works like this. What the circuit does is to sort it and put the embedding operations first. If you paint it you can see that the Hadamard gate is after the AmplitudeEmbedding. I think it would be more accurate to throw an error instead of ordering it but I don't know if it has another reason that I don't know.

Reply:
This was an intentional restriction at the time; the reason being was that PL originally did not support mixed states, and a mixed state would result if you applied an embedding to only some of the wires in the middle of the circuit!

However something we can definitely reconsider now that we have mixed state simulator support

Reply:
What I find curious is that AmpllitudeEmbedding and QubitStateVector work differently. What is the difference we want to show between them @josh146 ?

Reply:
How do you mean by work differently, @KetpuntoG? 🙂

Reply:
```python
import numpy as np
import pennylane as qml

dev = qml.device("default.qubit", wires=2)
@qml.qnode(dev)
def circuit():
    qml.Hadamard(wires = 0)
    qml.AmplitudeEmbedding([1,0,0,0], wires = [0,1])
    #qml.QubitStateVector([1,0,0,0], wires = [0,1])
    return qml.expval(qml.PauliZ(0))

circuit()
```

In this case only AmplitudeEmbedding will work.

Reply:
Ah, this is not intentional 😆 The `AmplitudeEmbedding` template is getting around our check that state preparation only occurs at the beginning of the circuit!

Reply:
Ah ok hahah makes sense.  Anyway I would change it to one that checks that no qubits have been used, which is kind of the idea I'm doing with this new AmplitudeEmbedding.